crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -137,35 +244,60 @@ impl Consensus {
                         txid: transaction.txid(),
                         error: err,
                     }
-                });
+                })?;
                 continue;
             }
+
             // Amount of all outputs
             let mut output_value = 0;
+
             for output in transaction.output.iter() {
                 Self::get_out_value(output, &mut output_value).map_err(|err| TransactionError {
                     txid: transaction.txid(),
                     error: err,
-                });
-                Self::validate_script_size(&output.script_pubkey).map_err(|err| TransactionError {
-                    txid: transaction.txid(),
-                    error: err,
-                });
+                })?;
+
+                Self::validate_script_size(&output.script_pubkey).map_err(|err| {
+                    TransactionError {
+                        txid: transaction.txid(),
+                        error: err,
+                    }
+                })?;
             }
+            // Validate Scripts
+            //
+            // NOTE: This should be verified before `Self::consume_utxos()` since its nature of removing the
+            // utxo from the map. Calling this on a removed or non-existent utxo will result in undefined behavior.
+            // (mostly will do nothing :D)
+            Self::validate_script(transaction, flags, &utxos).map_err(|err| TransactionError {
+                txid: transaction.txid(),
+                error: err,
+            })?;
```

Comment:
now we can do unit test on this, so we ensure that scripts are being validated.
NOTE: do the actually test on this